applications/GeoMechanicsApplication/custom_conditions/thermal_point_flux_condition.h

```diff
@@ -0,0 +1,68 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//
+
+#pragma once
+
+#include "custom_conditions/T_condition.h"
+#include "includes/serializer.h"
+
+namespace Kratos {
+
+template <unsigned int TDim, unsigned int TNumNodes>
+class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoThermalPointFluxCondition
+    : public GeoTCondition<TDim, TNumNodes> {
+public:
+    using GeometryType   = Geometry<Node>;
+    using PropertiesType = Properties;
+    using NodesArrayType = GeometryType::PointsArrayType;
+
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(GeoThermalPointFluxCondition);
+
+    GeoThermalPointFluxCondition();
+
+    GeoThermalPointFluxCondition(IndexType NewId, GeometryType::Pointer pGeometry);
+
+    GeoThermalPointFluxCondition(IndexType NewId,
+                            GeometryType::Pointer pGeometry,
+                            PropertiesType::Pointer pProperties);
+
+    ~GeoThermalPointFluxCondition() override;
+
+    Condition::Pointer Create(IndexType NewId,
+                              NodesArrayType const& rThisNodes,
+                              PropertiesType::Pointer pProperties) const override
+    {
+        return Kratos::make_intrusive<GeoThermalPointFluxCondition>(
+            NewId, this->GetGeometry().Create(rThisNodes), pProperties);
+    }
+
+protected:
+    void CalculateRHS(Vector& rRightHandSideVector,
+                      const ProcessInfo& CurrentProcessInfo) override;
+
+private:
+    friend class Serializer;
+
+    void save(Serializer& rSerializer) const override
+    {
+        KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, Condition)
```

Comment:
I would prefer to use the direct base class here (`GeoTCondition<TDim, TNumNodes>`), rather than an indirect one (`Condition`). The reason is that if the serialization of the direct base class becomes more involved, you would miss that here. In turn, that may lead to unexpected behavior when (de)serializing an instance of this class.
```suggestion
        using BaseType = GeoTCondition<TDim, TNumNodes>;
        KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, BaseType)
```